kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.coordinatedcommits;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.delta.storage.commit.CommitCoordinatorClient;
+
+/** Factory to get the correct {@link CommitCoordinatorClient} for a table */
```

Comment:
Add few more details like how connectors can register new commit cordinators and how this is used in the default coordinator client handler